internal/tool/multi/deployer.go

```diff
@@ -198,6 +179,68 @@ func newDeployer(ctx context.Context, deploymentId string, config *protos.AppCon
 	return d, nil
 }
 
+// computeGroups computes the colocation group information for the given
+// application.
+//
+// computeGroups places components into co-location groups based on the
+// colocate stanza in a config. For example, consider the following config.
+//
+//	colocate = [
+//	    ["A", "B", "C"],
+//	    ["D", "E"],
+//	]
+//
+// computeGroups creates a co-location group with components "A", "B", and
+// "C" and a co-location group with components "D" and "E". All other
+// components are placed in their own co-location group. We use the first
+// listed component as the co-location group name.
+func computeGroups(config *protos.AppConfig) (map[string]*group, error) {
+	groups := map[string]*group{}
+	ensureGroup := func(component string) *group {
+		if g, ok := groups[component]; ok {
+			return g
+		}
+		g := &group{
+			name:        component,
```

Comment:
I wonder if we should deterministically pick one of the component names as the group name? I worry about map iteration non-determinism or something else causing group names to fluctuate over time.

But perhaps this is a pre-existing problem that should be fixed in another change?